### PR TITLE
Progressive Web App feature + favicon improvements

### PR DESCRIPTION
**Pull request by timephy**
# Progressive Web App

Addition of the `static/manifest.json` allows the Mensa Website to be installed as a "Standalone App" on any smartphone ("Add to Home Screen"-Feature).

# Favicon

Favicon "source" is `svg/favicon.svg` which is used to generate (`bin/generate-favicon.sh`) the `favicon-xx.png`s.

### CODE REVIEW

File: static/manifest.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "TUM Mensa",
```
Comment (ackxolotl) [inline static/manifest.json]:
```suggestion
  "name": "Mensa-Homometer",
```

I think "Mensa-Homometer" would be a better name. "TUM Mensa" doesn't really reflect what this app is about?

File: static/manifest.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "short_name": "TUM Mensa",
```
Comment (ackxolotl) [inline static/manifest.json]:
```suggestion
  "short_name": "Mensa-Homometer",
```

File: homometer/templates/homometer/index.html [hunk 3]
```diff
@@ -9,9 +9,10 @@
 		<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no" />
 		<title>Mensa-Homometer</title>
 		<link href="{% static 'style.css' %}" rel="stylesheet" />
-		<link rel="apple-touch-icon" sizes="180x180" href="{% static 'apple-touch-icon.png' %}">
-		<link rel="icon" type="image/png" sizes="32x32" href="{% static 'favicon-32x32.png' %}">
-		<link rel="icon" type="image/png" sizes="16x16" href="{% static 'favicon-16x16.png' %}">
+		<link rel="apple-touch-icon" sizes="180x180" href="{% static 'favicon-180.png' %}">
```
Comment (ackxolotl) [inline homometer/templates/homometer/index.html]:
```suggestion
		<link rel="apple-touch-icon" sizes="180x180" href="{% static 'favicon-180.png' %}" />
```

My bad. Some tags had `/>` and some didn't. Can you unify this and add `/` to this line and the following two? :)

File: static/manifest.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  "icons": [
+    {
+      "src": "/images/favicon-192.png",
```
Comment (ackxolotl) [inline static/manifest.json]: This path is wrong: It should be `/static` and there is no favicon of size `192x192` (yet).

**Review by ackxolotl — CHANGES_REQUESTED** [review]
Thanks for your PR! I have some minor change requests. One big thing I noticed is that, although I like the color of the new favicon, it seems to be less recognizable than the old one (especially at `16x16`). Could you use [the old favicon](https://commons.wikimedia.org/wiki/File:AB-Rast.svg) or a better recognizable one?

**Comment by timephy** [comment]
Hey, thanks for the quick response!

I fixed all of the above issues. Changed to old icon with new background.
Fixed image path and html-typo.
Thanks for linking the original favicon symbol - I couldn't find it.

One more thing: I changed the PWA `short_name` to simply "Mensa", because the full name does not fit on mobile (see image).
image

Hope you find the changes satisfactory!

File: mensa/settings.py [hunk 5]
```diff
@@ -27,7 +27,7 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ["*"]
```
Comment (ackxolotl) [inline mensa/settings.py]:
```suggestion
ALLOWED_HOSTS = []
```

Is this change necessary? `ALLOWED_HOSTS` shouldn't matter in debug mode.

**Review by ackxolotl — COMMENTED** [review]
Looks good!

File: mensa/settings.py (hunk 5 above)
Comment (timephy) [reply mensa/settings.py]: No this is definitely not necessary, I didn't mean to commit/push this change... Therefore I reverted this change immediately. I wanted to access the website from my phone. Since in debug the server rejected the phone's (non-`localhost`) request I changed this line to make it work.

**Review by ackxolotl — APPROVED** [review]

**Comment by ackxolotl** [comment]
I've updated https://mensa.liste.party/. Hope it works as expected!
